### PR TITLE
Nitpicks related to platform/flash/zephyr review

### PR DESCRIPTION
**Pull request by larsewi**
- **chore: added .DS_Store to .gitignore**
- **fix: fixed error code check for boot and flash functions**
- **chore: fixed typo in comment**
- **refac: removed unnecessary variable definition and else clause**
- **docs: added comment explaining why we check if image is confirmed**
- **chore: fixed typo in comment**
- **style: moved constant to the left side in the if condition**
- **style: removed boolean comparison with false**

### CODE REVIEW

**Comment by larsewi** [comment]
> The following files are not properly formatted:
> platform/flash/zephyr/src/mender-flash.c

I'm not sure what is going on here. I seem to have the same version of `clang-format`:
```
$ clang-format --version
Ubuntu clang-format version 18.1.3 (1ubuntu1)
$ clang-format -i platform/flash/zephyr/src/mender-flash.c
$ git status
On branch flash
Your branch is up to date with 'origin/flash'.

nothing to commit, working tree clean
```

File: .gitignore [hunk 1]
```diff
@@ -9,4 +9,6 @@ CMakeCache.txt
 CMakeFiles/
 Makefile
 cmake_install.cmake
-libmender-mcu-client.a
\ No newline at end of file
+libmender-mcu-client.a
+
+.DS_Store
```
Comment (vpodzime) [inline .gitignore:14]: *noice* in the commit message. Interesting, I have never seen this file before. Should we add it everywhere across our projects? Or can you prevent if from being created?

File: platform/flash/zephyr/src/mender-flash.c [hunk 2]
```diff
@@ -40,8 +40,8 @@ mender_flash_open(char *name, size_t size, void **handle) {
     }
 
     /* Begin deployment with sequential writes */
-    if ((result = flash_img_init((struct flash_img_context *)*handle)) < 0) {
-        mender_log_error("flash_img_init failed (%d)", result);
+    if ((result = flash_img_init((struct flash_img_context *)*handle)) != 0) {
```
Comment (vpodzime) [inline platform/flash/zephyr/src/mender-flash.c:43]: Might as well do `(0 != (result = ...))` like we do in other places with other values.

File: platform/flash/zephyr/src/mender-flash.c [hunk 3]
```diff
@@ -126,20 +126,17 @@ mender_flash_abort_deployment(void *handle) {
 mender_err_t
 mender_flash_confirm_image(void) {
 
-    int          result;
-    mender_err_t ret = MENDER_OK;
-
     /* Validate the image if it is still pending */
     if (false == mender_flash_is_image_confirmed()) {
```
Comment (vpodzime) [inline platform/flash/zephyr/src/mender-flash.c:130]: Please also remove this boolean comparison, it looks lame.

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

File: .gitignore (hunk 1 above)
Comment (larsewi) [reply .gitignore:14]: Apparently you can:
```
$ defaults write com.apple.desktopservices DSDontWriteNetworkStores true
```
This file specifies folder icon positions, and other visual information in the macOS Finder app (i.e., the file explorer). I'm guessing that most mac users would like to not disable it, although, I just did. It seems to be created automatically once you open a directory in the Finder app.

Comment (larsewi) [reply .gitignore:14]: It does not create noice in the commit message. But it gets listed as one of those untracked files when you do `git status`.

File: platform/flash/zephyr/src/mender-flash.c (hunk 3 above)
Comment (larsewi) [reply platform/flash/zephyr/src/mender-flash.c]: It is pretty lame indeed

**Review by vpodzime — APPROVED** [review]
I'd add that comment about the weird MacOS file, but ACK either way. Thanks!

**Pull request merged**